.github/workflows/release.yaml

```diff
@@ -163,3 +188,16 @@ jobs:
           CertUtil -hashfile "dist/holochain-launcher-electron-${{ steps.version.outputs.APP_VERSION }}-setup.exe" SHA512
 
           gh release upload "v${{ steps.version.outputs.APP_VERSION }}" "dist/holochain-launcher-electron-${{ steps.version.outputs.APP_VERSION }}-setup.exe" --clobber
+  mark-as-prerelease:
```

Comment:
There is also a `preRelease` field in the ncipollo/release-action so this is obsolete. And if not necessary I prefer to not use github actions of repositories that are not well reviewed.

Reply:
I've changed it to just call GitHub API directly

Reply:
I don't understand why you would want to make an extra step for this (apart from that you may have just copied what ChatGPT told you - but I don't want to waste my time reviewing unfiltered ChatGPT output).
As I said in my comment there is a [field](https://github.com/ncipollo/release-action#action-inputs) in the `ncpiollo` action to have the created release be a preRelease so it is simply obsolete here.
And even the ncipollo action is not needed because if we look into the documentation of the tools we are already using first, the prerelease option is supported in electron builder by setting an environment variable and also that it publishes it directly is supported if we want that I think.
I just don't think that in the long term it is a good pattern to completely automate a release in the first place. We should always (at least rudimentarily) test the draft release before actually publishing it. And in those cases it is really handy based on my experience in other projects to have the tag be created only on publish instead of already when the draft release is being created (the latter is what the ncpiollo action does). Otherwise if you need to overwrite assets in the draft release with a fixed version before publishing you get into problems with the tag pointing to the wrong commit.

Reply:
Apologies for not stating this earlier, but the additional step will only be executed if all platforms from the previous step are successful. Unfortunately, I cannot use 'ncpiollo' due to issues with editing releases that have already been created. I completely agree that we should not automatically mark it as a 'release'. However, I believe it provides a solid basis for integrating an automated smoke test as an intermediate step (it is still "prerelease" not "release", than we could manually release before make a true "release")

> And in those cases it is really handy based on my experience in other projects to have the tag be created only on publish instead of already when the draft release is being created (the latter is what the ncpiollo action does). Otherwise if you need to overwrite assets in the draft release with a fixed version before publishing you get into problems with the tag pointing to the wrong commit.

Can you expand on this? Because I can't see the difference how I made a last pre-release manually and with this automation

Reply:
> I cannot use 'ncpiollo' due to issues with editing releases that have already been created.

What do you mean by that? I don't mean to use ncipollo to edit the release after creation, I mean to have it be a prerelease from the beginning. Or is this not possible for draft releases?

> Can you expand on this?
What I mean with that is that the ncipollo action creates a tag upon (draft) release creation based on the package.json version. But if we want to overwrite stuff in the draft release before publishing as a pre-release there is already a tag at the previous commit.
But I understand now that we differ mainly in how we intend to approach draft/pre-release/release. So far (in the old launcher repository) I have managed it in the way that I did not publish any publicly visible release (be it pre-release or release) at all before doing some rudimentary testing on it. And my notion of a "release" vs. a "pre-release" is that a "release" is one that's considered end-user ready and so I never even made a "release" of a launcher so far, I always only made "pre-releases" to indicate that we are still far from stability reagarding end-user-usability.